Yoast/wordpress-seo
# P3-203 create presenter for enhanced Slack sharing

## Context


* We want to add some tags (originally introduced by Twitter) in the `<head>` section of single posts and pages to display "Written by John Doe" and "Est. reading time x minutes" when the post is shared on Slack:
![slack sharing](https://user-images.githubusercontent.com/15989132/95571907-7652b200-0a29-11eb-8a03-a968e3cc22c3.png)
This must happens only if the feature introduced in #16164 is enabled.

## Summary


This PR can be summarized in the following changelog entry:

* Adds enhanced Slack sharing tags on posts and pages.

## Relevant technical choices:

* The array which will be cycled to display the enhanced data can be filtered by hooking on 'wpseo_enhanced_data'
* Consistently with Twitter and OpenGraph tags, the filter `wpseo_output_enhanced_slack_data` has been introduced to override the DB option
* Since the WP core has already strings for "%minute(s)", we opted for using them instead of introducing a duplicate in our language files

## Test instructions

This PR can be tested by following these steps:

* Make sure that the "Enhanced Slack sharing" toggle in SEO→General, "Features" tab, is enabled.
* visit a single post or a single page in the frontend and inspect the source.
* check that just before the schema there are tags like these ones:
```
	<meta name="twitter:label1" value="Written by">
	<meta name="twitter:data1" value="Agatha Christie">
	<meta name="twitter:label2" value="Est. reading time">
	<meta name="twitter:data2" value="29 minutes">
```
* check for different posts/pages with different length of the content.
* also check that these are not displayed on archive pages
* disable the feature toggle and save the settings
* check that the tags are not displayed anymore.

## UI changes

## Documentation

## Quality assurance


Fixes [P3-203]
